### DISCUSSION TITLE
Valid app Identifier Missing - Phone auth - react-native-firebase

### DISCUSSION THREAD

**Discussion by sourdragon** [invertase/react-native-firebase#7094] [Q&A]
# :cry: [Error](https://i.imgur.com/TD1HbIz.png) I'm getting

> Console Warning Possible Unhandled Promise Rejection (id: 0): Error:
> [auth/missing-client-identifier]  This request is missing a valid app
> identifier, meaning that Play Integrity checks, SafetyNet checks,  and
> reCAPTCHA checks were unsuccessful. Please try again, or check the
> logcat for more details.  NativeFirebaseError: Iauth/
> missing-client-identifier] This request is missing a valid app
> identifier,  meaning that ***Play Integrity checks, SafetyNet checks,
> and reCAPTCHA checks*** were unsuccessful.  Please try again, or check
> the logcat for more details.

# :1234: Steps to reproduce

- npx react-native@latest init Example && cd Example

- cd android && ./gradlew signingReport

- // download the google-services.json with the sha1 & sha256 to the /android/app
- [Add these lines](https://rnfirebase.io/#configure-firebase-with-android-credentials)
- npm i @react-native-firebase/app @react-native-firebase/auth
- copy paste this [code](https://rnfirebase.io/auth/phone-auth) into your App.tsx

# :mag:  Stuff I did to get rid of the issue(to no avail)
- added implementation 'com.google.android.play:integrity:1.1.0' to android/app/build.gradle

- Enabled API and Services

- Added SHA-1 and SHA-256 codes

- {PROJECT_ID}.firebase.app is allowlisted at firebase authentication settings.

**Comment by sleekLancelot** [comment]
Currently facing this issue.
@sourdragon Hi, were you able to resolve this?

**Reply by Meet7056** [reply]
can you solved it now??

**Reply by sleekLancelot** [reply]
Hi, in my case some of the services were protected by app check, so I used app check in the dev environment. 
Also In prod(app downloaded from google play or apple store) I didn't face this error since the app was already signed.

**Comment by Meet7056** [comment]
i am facing the same issue... please give me the solution of this ...
